src/napari/_vispy/utils/zoom.py

```diff
@@ -0,0 +1,81 @@
+"""Vispy zoom box overlay."""
+
+from __future__ import annotations
+
+import numpy as np
+
+from napari.viewer import ViewerModel
+
+
+def _get_dim_info(
+    mins: np.ndarray, maxs: np.ndarray
+) -> tuple[np.ndarray, np.ndarray]:
+    """Calculate center and difference for single dimension."""
+    center = (mins + maxs) / 2
+    spread = maxs - mins
+    spread[spread == 0] = 1.0  # avoid division by zero
+    return center, spread
+
+
+def _get_data_extents(
+    data_positions: tuple[tuple[float, ...], tuple[float, ...]],
+    displayed: tuple[int, ...],
+) -> tuple[np.ndarray, np.ndarray]:
```

Comment:
Similarly, I would rename this to something like `_zoombox_corners_world_to_displayed` (assuming I understand correctly this function)